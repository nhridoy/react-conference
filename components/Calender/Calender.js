import Link from "next/link";
const Calender = ({ conferences }) => {
  const data = {};
  for (const key in conferences) {
    if (Object.hasOwnProperty.call(conferences, key)) {
      const element = conferences[key];
      data[element.year] = [];
    }
  }
  for (const key in conferences) {
    if (Object.hasOwnProperty.call(conferences, key)) {
      const element = conferences[key];
      data[element.year].push(element);
    }
  }

  return (
    <div className="rounded-xl border overflow-x-auto">
      <table className="w-full">
        <tbody>
          {Object.keys(data).map((item, index) => (
            <tr key={index}>
              <th className="p-10 md:p-0 w-36 h-36 border">{item}</th>
              {data[item].map((schedule, i) => (
                <td
                  key={i}
                  className="p-10 md:p-0 w-36 h-36 border text-center"
                >
                  <span className="font-bold">{schedule.startDate}</span>
                  <Link
                    href={`/${schedule.id}`}
                    className="p-1 bg-blue-600/30 border-blue-700 border rounded-lg text-blue-700 flex flex-col m-2"
                  >
                    <span className="font-semibold">{schedule.name}</span>
                    <span className="text-sm">
                      {schedule?.locations?.length &&
                        schedule?.locations[0]?.city}
                    </span>
                  </Link>
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Calender;
