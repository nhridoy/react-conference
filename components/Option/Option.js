import { BsArrowDownUp } from "react-icons/bs";
const Option = ({ name, id }) => {
  return (
    <div className="border rounded-md flex items-center gap-4 p-4 bg-white hover:bg-yellow-400 cursor-pointer group font-semibold drop-shadow-lg">
      <div className="bg-white p-3 rounded-md cursor-fancy">
        <BsArrowDownUp className="text-lg text-yellow-400 " />
      </div>
      <span className="group-hover:text-white">{name}</span>
    </div>
  );
};

export default Option;
