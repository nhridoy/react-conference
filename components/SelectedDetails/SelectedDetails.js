import Image from "next/image";

const SelectedDetails = ({
  name,
  about,
  day,
  firstName,
  lastName,
  image,
  company,
}) => {
  return (
    <div className="bg-white p-2 md:p-4 grid grid-cols-5 gap-2 items-center rounded-md">
      <div className="col-span-2 md:col-span-1">
        {image ? (
          <Image
            className="rounded-lg w-28 h-28 object-contain"
            src={image}
            height={100}
            width={100}
            alt={name}
          />
        ) : (
          <b className="rounded-lg w-28 h-28">{day}</b>
        )}
      </div>
      <div className="col-span-3 md:col-span-4 flex flex-col gap-4">
        <div className="flex flex-col md:flex-row justify-between md:items-end">
          <b className="font-bold">{name}</b>
          <p className="text-sm text-gray-700">{company}</p>
        </div>
        <p className="text-justify">{about}</p>
      </div>
    </div>
  );
};

export default SelectedDetails;
