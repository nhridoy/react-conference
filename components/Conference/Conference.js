import Option from "../Option/Option";
import { SortableContainer, SortableElement } from "react-sortable-hoc";
import { useState } from "react";
import { arrayMoveImmutable } from "array-move";

const SortableItem = SortableElement(({ value }) => <Option name={value} />);

const SortableList = SortableContainer(({ items }) => {
  return (
    <ul className=" flex flex-col gap-4">
      {items.map((value, index) => (
        <SortableItem key={`item-${value}`} index={index} value={value} />
      ))}
    </ul>
  );
});

const Conference = () => {
  const [items, setItems] = useState([
    "Item 1",
    "Item 2",
    "Item 3",
    "Item 4",
    "Item 5",
    "Item 6",
  ]);
  const onSortEnd = ({ oldIndex, newIndex }) => {
    setItems((items) => arrayMoveImmutable(items, oldIndex, newIndex));
  };

  return (
    <>
      <h2 className="font-bold text-4xl mb-4 mt-16">Conference info</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Fuga, cumque.
      </p>
      <div className="grid grid-cols-1 md:grid-cols-3 gap-5 mt-8">
        <div className="col-span-1">
          <SortableList items={items} onSortEnd={onSortEnd} />
        </div>
        <div className="md:col-span-2"></div>
      </div>
    </>
  );
};

export default Conference;
