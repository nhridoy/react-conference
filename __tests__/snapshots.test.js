import Calender from "../components/Calender/Calender";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import allConference, { singleConference } from "../data/data";
import Conference from "../components/Conference/Conference";
import Option from "../components/Option/Option";
import SelectedDetails from "../components/SelectedDetails/SelectedDetails";

afterAll(() => {
  cleanup();
});

it("all conference table", () => {
  const { container } = render(<Calender conferences={allConference} />);
  expect(container).toMatchSnapshot();
});

it("single conference", () => {
  const { container } = render(<Conference data={singleConference} />);
  expect(container).toMatchSnapshot();
});

it("single conference details", () => {
  const { container } = render(
    <SelectedDetails
      name={
        singleConference?.organizers?.name ||
        singleConference?.organizers?.location?.name
      }
      about={
        singleConference?.organizers?.about ||
        singleConference?.organizers?.location?.about
      }
      firstName={singleConference?.organizers?.firstName}
      lastName={singleConference?.organizers?.lastName}
      image={singleConference?.organizers?.image?.url}
      day={singleConference?.organizers?.day}
      company={
        singleConference?.organizers?.company ||
        singleConference?.organizers?.location?.city
      }
    />
  );
  expect(container).toMatchSnapshot();
});
