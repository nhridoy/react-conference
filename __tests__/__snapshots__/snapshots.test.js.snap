// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all conference table 1`] = `
<div>
  <div
    class="rounded-xl border overflow-x-auto"
  >
    <table
      class="w-full"
    >
      <tbody>
        <tr>
          <th
            class="p-10 md:p-0 w-36 h-36 border"
          >
            undefined
          </th>
          <td
            class="p-10 md:p-0 w-36 h-36 border text-center"
          >
            <span
              class="font-bold"
            />
            <a
              class="p-1 bg-blue-600/30 border-blue-700 border rounded-lg text-blue-700 flex flex-col m-2"
              href="/undefined"
            >
              <span
                class="font-semibold"
              />
              <span
                class="text-sm"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`single conference 1`] = `
<div>
  <h2
    class="font-bold text-4xl mb-4 mt-16"
  >
    Conference info 
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Fuga, cumque.
  </p>
  <div
    class="grid grid-cols-1 md:grid-cols-3 gap-5 mt-8"
  >
    <div
      class="col-span-1"
    >
      <div
        class="flex flex-col gap-4"
        data-rbd-droppable-context-id="0"
        data-rbd-droppable-id="characters"
      >
        <div
          aria-describedby="rbd-hidden-text-0-hidden-text-0"
          data-rbd-drag-handle-context-id="0"
          data-rbd-drag-handle-draggable-id="item0"
          data-rbd-draggable-context-id="0"
          data-rbd-draggable-id="item0"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="bg-yellow-400 border rounded-md flex items-center gap-4 p-4 hover:bg-yellow-400 cursor-fancy group font-semibold drop-shadow-lg"
          >
            <div
              class="bg-white p-3 rounded-md cursor-fancy"
            >
              <svg
                class="text-lg text-yellow-400 "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.5 15a.5.5 0 0 0 .5-.5V2.707l3.146 3.147a.5.5 0 0 0 .708-.708l-4-4a.5.5 0 0 0-.708 0l-4 4a.5.5 0 1 0 .708.708L11 2.707V14.5a.5.5 0 0 0 .5.5zm-7-14a.5.5 0 0 1 .5.5v11.793l3.146-3.147a.5.5 0 0 1 .708.708l-4 4a.5.5 0 0 1-.708 0l-4-4a.5.5 0 0 1 .708-.708L4 13.293V1.5a.5.5 0 0 1 .5-.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <span
              class="text-white group-hover:text-white"
            >
              Data
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="md:col-span-2 bg-gray-100 rounded-lg p-2 md:p-10 flex flex-col gap-4 md:max-h-screen overflow-y-scroll"
    />
  </div>
</div>
`;

exports[`single conference details 1`] = `
<div>
  <div
    class="bg-white p-2 md:p-4 grid grid-cols-5 gap-2 items-center rounded-md"
  >
    <div
      class="col-span-2 md:col-span-1"
    >
      <b
        class="rounded-lg w-28 h-28"
      />
    </div>
    <div
      class="col-span-3 md:col-span-4 flex flex-col gap-4"
    >
      <div
        class="flex flex-col md:flex-row justify-between md:items-end"
      >
        <b
          class="font-bold"
        />
        <p
          class="text-sm text-gray-700"
        />
      </div>
      <p
        class="text-justify"
      />
    </div>
  </div>
</div>
`;
